tests/components/husqvarna_automower/test_init.py

```diff
@@ -127,21 +127,50 @@ async def test_websocket_not_available(
     caplog: pytest.LogCaptureFixture,
     freezer: FrozenDateTimeFactory,
 ) -> None:
-    """Test trying reload the websocket."""
-    mock_automower_client.start_listening.side_effect = HusqvarnaWSServerHandshakeError(
-        "Boom"
-    )
-    await setup_integration(hass, mock_config_entry)
-    assert "Failed to connect to websocket. Trying to reconnect: Boom" in caplog.text
-    assert mock_automower_client.auth.websocket_connect.call_count == 1
-    assert mock_automower_client.start_listening.call_count == 1
-    assert mock_config_entry.state is ConfigEntryState.LOADED
-    freezer.tick(timedelta(seconds=2))
-    async_fire_time_changed(hass)
-    await hass.async_block_till_done()
-    assert mock_automower_client.auth.websocket_connect.call_count == 2
-    assert mock_automower_client.start_listening.call_count == 2
-    assert mock_config_entry.state is ConfigEntryState.LOADED
+    """Test trying to reload the websocket."""
+    # Patch DEFAULT_RECONNECT_TIME to 0 for the test
+    with patch(
+        "homeassistant.components.husqvarna_automower.coordinator.DEFAULT_RECONNECT_TIME",
+        new=0,
+    ):
+        # Simulate a WebSocket handshake error
+        mock_automower_client.auth.websocket_connect.side_effect = (
```

Comment:
I think we should customize the side effect with a function so that the first task raises the exception but then in the second task we block, eg by waiting for an `asyncio.Event` or `asyncio.Future`. Otherwise there'll be a fast loop creating tasks here.

Reply:
Sorry, I don't understand what you mean. Can you try to explain that with different words. Or would it also possible to not patch the `DEFAULT_RECONNECT_TIME` not to zero, to avoid the changes?

Reply:
Here's an example of how to set a side effect that blocks until an event (`listen_block`) is set by the test.

https://github.com/home-assistant/core/blob/866ffcf639d3be0f4e6a35884b41fa3e6bff83b2/tests/components/zwave_js/conftest.py#L500-L526